WooCommerce/Classes/ViewRelated/Hub Menu/HubMenuViewModel.swift

```diff
@@ -73,16 +73,16 @@ final class HubMenuViewModel: ObservableObject {
     private let posEligibilityChecker: POSEligibilityCheckerProtocol
     private let inboxEligibilityChecker: InboxEligibilityChecker
 
-    // TODO:
-    // Is this the right place to instantiate the product provider and use property injection?
     private(set) lazy var posItemProvider: POSItemProvider = {
         let storageManager = ServiceLocator.storageManager
         let siteID = ServiceLocator.stores.sessionManager.defaultSite?.siteID ?? 0
+        let credentials = ServiceLocator.stores.sessionManager.defaultCredentials
```

Comment:
> These are available in the HubMenuViewModel.credentials property now, as is the siteID.
Oh nice, updated on [1638abd](https://github.com/woocommerce/woocommerce-ios/pull/13116/commits/1638abdddec131f39ae135779afd78a4c010dc88)

> It would probably be better not to have this as a lazy var as it often leads to issues...

I see 🤔 , for this I believe we've used the same pattern as other view models within the HubMenu. I'll explore this separately and see if/how we can remove the lazy instantiation.

Reply:
Since it gets injected to `PointOfSaleEntryPointView`, and we're unlikely to write tests for that, you could build the default one in the `PointOfSaleEntryPointView.init` function, which would make it a lot simpler.